src/js/controllers/imageCtrl-test.js

```diff
@@ -0,0 +1,142 @@
+                $provide.service('images', function($q){
+                    this.loadPics = sinon.stub().returns($q.resolve(mockImageObject));
+                });
+
+            }));
+            beforeEach(inject(function (_$rootScope_, _$controller_, _images_, _$q_, _$log_) {
+                $rootScope = _$rootScope_.$new();
+                $controller = _$controller_;
+                images = _images_;
+                $q = _$q_;
+                $log = _$log_;
+                vm = $controller('imagesCtrl');
+
+                images.loadPics.returns($q.resolve(mockImageObject));
+                $rootScope.$apply();
+            }));
+
+            describe('returned flickr images object from http call', function(){
+
+                it('should return an object', function () {
+                    expect(angular.isObject(vm.images)).to.equal(true);
+                });
+
+                it('parent object should contain a photos object', function () {
+                    expect(angular.isObject(vm.images.photos)).to.equal(true);
+                });
+
+                it('photos object should include an photo object', function(){
+                    expect(angular.isObject(vm.images.photos.photo)).to.equal(true);
+                });
+
+            });
+
+            describe('individual photo object', function(){
+                var photo;
+
+                beforeEach(function(){
+                    photo = vm.images.photos.photo;
+                });
+
+                it('check for neccesary attibutes', function(){
+                    angular.forEach(photo, function(value){
+                        expect(value.farm).to.exist();
+                        expect(value.server).to.exist();
+                        expect(value.id).to.exist();
+                        expect(value.owner).to.exist();
+                        expect(value.secret).to.exist();
+                    });
+                });
+
+                it('check for neccesary attibutes to not be undefined ', function(){
+                    angular.forEach(photo, function(value){
+                        expect(value.farm).not.to.be.undefined();
+                        expect(value.server).not.to.be.undefined();
+                        expect(value.id).not.to.be.undefined();
+                        expect(value.owner).not.to.be.undefined();
+                        expect(value.secret).not.to.be.undefined();
+                    });
+                });
+
+                it('check for neccesary attibutes to not be null ', function(){
+                    angular.forEach(photo, function(value){
+                        expect(value.farm).not.to.be.null();
+                        expect(value.server).not.to.be.null();
+                        expect(value.id).not.to.be.null();
+                        expect(value.owner).not.to.be.null();
+                        expect(value.secret).not.to.be.null();
+                    });
+                });
```

Comment:
The 3 tests are ultimately testing different variations of the same thing. I agree that a check to ensure that the values exist is useful, however, I feel 2 of these are overkill. :smile: 

If you'd like test something in addition to the existence, you could check the specific values of one (or more) of the returned data. And if you plan on doing any data manipulation, you could verify that the values match the appropriate format.